### PR TITLE
Trib 150: bug fix saves "nullvalue" for blank adverb/preposition

### PR DESCRIPTION
**Pull request by mrsbluerose**
- updates applyPhraseToUser to save "nullvalue" in blank adverbs or prepositions in newly added phrases
- updates findPreviouslyApprovedPhraseId comparisons to check for "nullvalue" when building a phrase to search the phrase repo

### CODE REVIEW

**Review by haxwell — CHANGES_REQUESTED** [review]
We don't have any tests for applyPhraseToUser which take null/empty values. Can you add some?

File: src/test/java/com/savvato/tribeapp/services/PhraseServiceImplTest.java [hunk 1]
```diff
@@ -194,11 +193,41 @@ public void testApplyPhraseToUserWhenPhraseDoesNotExistInToBeReviewed() {
...
+        tbrSaved.setId(1L);
+        tbrSaved.setHasBeenGroomed(false);
+        tbrSaved.setAdverb(testAdverbConverted);
+        tbrSaved.setVerb(testVerb);
+        tbrSaved.setPreposition(testPreposition);
+        tbrSaved.setNoun(testNoun);
+
+        // Should return false if the phrase has not been seen before
+        Mockito.when(toBeReviewedRepository.save(any())).thenReturn(tbrSaved);
+        assertFalse(phraseService.applyPhraseToUser(user1.getId(),testAdverb,testVerb,testPreposition, testNoun));
```
Comment (haxwell) [inline src/test/java/com/savvato/tribeapp/services/PhraseServiceImplTest.java:223]: Separate line 223 into 1) the call to `applyPhraseToUser` and 2) the assertion on its return value. The reason is that it's harder to see the 'when' (phraseService.applyPhraseToUser()) section of this test when it is wrapped in a 'then' function (assertFalse)

See https://martinfowler.com/bliki/GivenWhenThen.html

**Comment by haxwell** [comment]
Also, you have a test for a blank Adverb, but not for a blank Preposition.

File: src/test/java/com/savvato/tribeapp/services/PhraseServiceImplTest.java (hunk 1 above)
Comment (mrsbluerose) [reply src/test/java/com/savvato/tribeapp/services/PhraseServiceImplTest.java]: separated

**Review by haxwell — APPROVED** [review]

**Pull request merged**